internal/testrunner/runners/system/runner.go

```diff
@@ -261,33 +264,82 @@ func (r *runner) Run(ctx context.Context, options testrunner.TestOptions) ([]tes
+		PackageName:        r.options.TestFolder.Package,
+		DataStream:         r.options.TestFolder.DataStream,
+		PolicyName:         policyName,
+		Variant:            variantName,
 		RunTearDown:        r.options.RunTearDown,
 		RunTestsOnly:       r.options.RunTestsOnly,
 		RunSetup:           r.options.RunSetup,
 	}
 }
 
-func (r *runner) createServiceInfo(serviceOptions servicedeployer.FactoryOptions) (servicedeployer.ServiceInfo, error) {
+func (r *runner) createServiceOptions(variantName string) servicedeployer.FactoryOptions {
+	return servicedeployer.FactoryOptions{
+		Profile:                r.options.Profile,
+		PackageRootPath:        r.options.PackageRootPath,
+		DataStreamRootPath:     r.dataStreamPath,
+		DevDeployDir:           DevDeployDir,
+		Variant:                variantName,
+		Type:                   servicedeployer.TypeTest,
+		StackVersion:           r.stackVersion.Version(),
+		PackageName:            r.options.TestFolder.Package,
+		DataStream:             r.options.TestFolder.DataStream,
+		RunTearDown:            r.options.RunTearDown,
+		RunTestsOnly:           r.options.RunTestsOnly,
+		RunSetup:               r.options.RunSetup,
+		DeployIndependentAgent: r.options.RunIndependentElasticAgent,
+	}
+}
+
+func (r *runner) createAgentInfo(policy *kibana.Policy) (agentdeployer.AgentInfo, error) {
+	var info agentdeployer.AgentInfo
+
+	info.Name = r.options.TestFolder.Package
+	info.Logs.Folder.Agent = ServiceLogsAgentDir
+	info.Test.RunID = createTestRunID()
+
+	folderName := fmt.Sprintf("agent-%s", r.options.TestFolder.Package)
+	if r.options.TestFolder.DataStream != "" {
+		folderName = fmt.Sprintf("%s-%s", folderName, r.options.TestFolder.DataStream)
+	}
+
+	dirPath, err := agentdeployer.CreateServiceLogsDir(r.locationManager, folderName)
+	if err != nil {
+		return agentdeployer.AgentInfo{}, fmt.Errorf("failed to create service logs dir: %w", err)
+	}
+	info.Logs.Folder.Local = dirPath
+
+	info.Policy.Name = policy.Name
+	info.Policy.ID = policy.ID
+
+	return info, nil
+}
+
+func (r *runner) createServiceInfo() (servicedeployer.ServiceInfo, error) {
 	var svcInfo servicedeployer.ServiceInfo
 	svcInfo.Name = r.options.TestFolder.Package
 	svcInfo.Logs.Folder.Local = r.locationManager.ServiceLogDir()
 	svcInfo.Logs.Folder.Agent = ServiceLogsAgentDir
 	svcInfo.Test.RunID = createTestRunID()
 
-	outputDir, err := servicedeployer.CreateOutputDir(r.locationManager, svcInfo.Test.RunID)
-	if err != nil {
-		return servicedeployer.ServiceInfo{}, fmt.Errorf("could not create output dir for terraform deployer %w", err)
+	if r.options.RunTearDown || r.options.RunTestsOnly {
+		logger.Debug("Skip creating output directory")
+	} else {
+		outputDir, err := servicedeployer.CreateOutputDir(r.locationManager, svcInfo.Test.RunID)
+		if err != nil {
+			return servicedeployer.ServiceInfo{}, fmt.Errorf("could not create output dir for terraform deployer %w", err)
+		}
+		svcInfo.OutputDir = outputDir
```

Comment:
Ensure that this folder is just created with --setup or regular testing commands.

The OutputDir and RunID values will be retrieved from ServiceState in case testing is run in stages (e.g. --no-provision)